src/sprite.c

```diff
@@ -54,8 +54,10 @@ void sprite_free(struct sact_sprite *sp)
 	scene_unregister_sprite(&sp->sp);
 	gfx_delete_texture(&sp->texture);
 	gfx_delete_texture(&sp->text.texture);
-	if (sp->plugin)
+	if (sp->plugin && sp->plugin->free) {
+		sp->plugin->free(sp->plugin);
 		LIST_REMOVE(sp, entry);
+	}
```

Comment:
Shouldn't the `LIST_REMOVE` happen here regardless of whether there's a `free` function defined?

Reply:
Good catch! Fixed.